### DISCUSSION TITLE
Support image clipboard contents

### DISCUSSION THREAD

**Discussion by tobiasbueschel** [espanso/espanso#1267] [Feature requests and Ideas]
**Describe the bug**
Hi all, I noticed that form matches do not seem to support clipboard inserts if the clipboard contains an image.

**To Reproduce**
1. Use the following match trigger
```yml
  - trigger: ":flight"
    form: |
      Hi [[name]],

      Does this flight itinerary work for you?

      {{clipboard}}

      Cheers
    form_fields:
      name:
        multiline: false
        default: "John"
```

2. Copy an image to your clipboard using e.g., macOS screenshot functionality
3. Executed text expansion `:flight` and notice that Espanso throws an error

**Expected behavior**
Picture is expanded in form dialog preview and inserted correct with the rest of the expansion.

**Screenshots**
<img width="606" alt="image" src="https://user-images.githubusercontent.com/13087421/160354928-b6ec85b9-24d5-4bdf-8a4d-fe0384d2a186.png">

**Logs**
```
16:12:25 [worker(82380)] [INFO] monitoring the status of the daemon process
16:12:25 [worker(82380)] [INFO] monitoring the status of secure input
16:12:29 [worker(82380)] [WARN] detected inconsistent modifier state for keys [(Meta, 55)], sending compensating events...
16:12:36 [worker(82380)] [WARN] rendering was aborted by extension: form, on var: form1
16:12:41 [worker(82380)] [WARN] extension 'clipboard' on var: 'clipboard' reported an error: clipboard provider returned error
16:12:41 [worker(82380)] [ERROR] error during rendering: rendering error

Caused by:
    clipboard provider returned error
16:12:58 [worker(82380)] [WARN] rendering was aborted by extension: form, on var: form1
16:13:03 [worker(82380)] [WARN] extension 'clipboard' on var: 'clipboard' reported an error: clipboard provider returned error
16:13:03 [worker(82380)] [ERROR] error during rendering: rendering error

Caused by:
    clipboard provider returned error
```

**Setup information**
- OS: macOS 12.3
- Version: 2.1.4-beta

**Comment by federico-terzi** [comment]
Hey @tobiasbueschel,

Thank you for the feedback! This is happening because Espanso currently doesn't support image clipboard contents, but I agree it would be a nice feature to have. We'll investigate :)

Cheers :)
